support-services/src/main/scala/com/gu/helpers/WebServiceHelper.scala

```diff
@@ -51,19 +65,32 @@ trait WebServiceHelper[Error <: Throwable] {
    */
   private def request[A](rb: Request.Builder)(implicit decoder: Decoder[A], errorDecoder: Decoder[Error], ctag: ClassTag[A]): Future[A] = {
     val req = wsPreExecute(rb).build()
-    SafeLogger.debug(s"Issuing request ${req.method} ${req.url}")
+    SafeLogger.info(s"Issuing request ${req.method} ${req.url}")
     for {
       response <- httpClient(req)
-    } yield {
-      val responseBody = response.body.string()
-      SafeLogger.debug(responseBody)
-      decode[A](responseBody) match {
-        case Left(err) => throw decodeError(responseBody).right.getOrElse(
-          WebServiceHelperError[A](response.code(), responseBody)
-        )
-        case Right(value) => value
+      decodedResponse <- {
+        val code = response.code().toString
+        val failableDecodedResponse = for {
+          body <- Option(response.body).toRight(WebServiceHelperError(code, "", "no body")).toTry
+          contentType <- Option(body.contentType()).toRight(WebServiceHelperError(code, "", "no content type")).toTry
+          bodyString <- Try(body.string())
+          _ = SafeLogger.info(s"response $code body: ${bodyString.length} bytes")
+          _ <- if (contentType.`type`() == "application" && contentType.subtype() == "json") Success(())
+          else Failure(WebServiceHelperError(code, bodyString, s"wrong content type"))
+          result <- code.splitAt(1) match {
```

Comment:
I think this match block could also be a separate function, there's a lot going on in this `for` block